test/org/zaproxy/zap/extension/pscanrulesBeta/ServletParameterPollutionScannerUnitTest.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2016 The ZAP development team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.pscanrulesBeta;
+
+import org.junit.Test;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ServletParameterPollutionScannerUnitTest extends PassiveScannerTest {
+
+    public static final String URI = "http://www.example.com/test/";
+
+    @Override
+    protected ServletParameterPollutionScanner createScanner() {
+        return new ServletParameterPollutionScanner();
+    }
+
+    @Test
+    public void GivenNoForms_WhenScanHttpResponseReceive_ThenNoAlertsRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("");
+        scanHttpResponseReceive(msg);
+        assertNumberOfAlertsRaised(0);
+    }
+
+    @Test
+    public void GivenFormWithActionParameter_WhenScanHttpResponseReceive_ThenNoAlertsRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form action='ActionMan'>");
+        scanHttpResponseReceive(msg);
+        assertNumberOfAlertsRaised(0);
+    }
+
+    @Test
+    public void GivenFormWithNoActionParameter_WhenScanHttpResponseReceive_ThenAlertRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form />");
+        scanHttpResponseReceive(msg);
+        assertNumberOfAlertsRaised(1);
+    }
+
+    @Test
+    public void GivenFormWithValuelessActionParameter_WhenScanHttpResponseReceive_ThenAlertRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form action />");
+        scanHttpResponseReceive(msg);
+        assertNumberOfAlertsRaised(1);
+    }
+
+    @Test
+    public void GivenFormWithEmptyActionParameter_WhenScanHttpResponseReceive_ThenAlertRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form action='' />");
+        scanHttpResponseReceive(msg);
+        assertNumberOfAlertsRaised(1);
+    }
+
+    @Test
+    public void GivenFormWithEmptyAndPopulatedActionParameters_WhenScanHttpResponseReceive_ThenAlertRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form action action='ActionMan' />");
```

Comment:
I believe this is a defect, browsers will take the first attribute (which is empty) but the scanner treats this is fine.